### PR TITLE
[im-2290] Instrument timeout exceptions

### PR DESCRIPTION
**Pull request by anujc4**
## DESCRIPTION

JIRA Ticket: [im-2290](https://deliveroo.atlassian.net/browse/im-2290)

CHANGELOG:

Instrument timeout exceptions. There are a few tests failing on the PR, but these are unrelated to my changes and have been failing on the CI for some time now.

### CODE REVIEW

File: lib/routemaster/middleware/metrics.rb [hunk 1]
```diff
@@ -22,6 +22,9 @@ def call(request_env)
           rescue Routemaster::Errors::BaseError => e
             increment_response_count(response_tags(e.env))
             raise e
+          rescue Faraday::TimeoutError => e
```
Comment (smnalex) [inline lib/routemaster/middleware/metrics.rb:25]: (If the circuit is open we won't report any metrics)

Comment (smnalex) [inline lib/routemaster/middleware/metrics.rb:25]: Should we add connection too?

Comment (anujc4) [reply lib/routemaster/middleware/metrics.rb:25]: Yep, but we would know how many timeouts we received before the breaker was tripped. We anyway don't push any metric once a circuit trips. We would have to record circuitbox metrics for that information.

Comment (anujc4) [reply lib/routemaster/middleware/metrics.rb:25]: Wdym? `Routemaster::Errors::BaseError` and `Faraday::TimeoutError` are the only exceptions that are monitored by CircuitBox

**Review by smnalex — APPROVED** [review]

File: lib/routemaster/middleware/metrics.rb (hunk 1 above)
Comment (smnalex) [reply lib/routemaster/middleware/metrics.rb:25]: I meant `Faraday::ConnectionFailed`, but sorry, forgot that you made this change only for CB.

**Pull request merged**